internal/lsp/lsproto/_generate/generate.mjs

```diff
@@ -26,524 +26,805 @@ if (!fs.existsSync(metaModelPath)) {
+            ) {
+                return { name: "[2]uint32", needsPointer: false };
             }
-            if (line.startsWith("@proposed")) {
-                line = "Proposed.\n//";
+
+            throw new Error("Unsupported tuple type: " + JSON.stringify(type));
+        }
+
+        case "stringLiteral": {
+            const typeName = `StringLiteral${titleCase(type.value)}`;
+            typeInfo.literalTypes.set(String(type.value), typeName);
+            return { name: typeName, needsPointer: false };
+        }
+
+        case "integerLiteral": {
+            const typeName = `IntegerLiteral${type.value}`;
+            typeInfo.literalTypes.set(String(type.value), typeName);
+            return { name: typeName, needsPointer: false };
+        }
+
+        case "booleanLiteral": {
+            const typeName = `BooleanLiteral${type.value ? "True" : "False"}`;
+            typeInfo.literalTypes.set(String(type.value), typeName);
+            return { name: typeName, needsPointer: false };
+        }
+        case "literal":
+            if (type.value.properties.length === 0) {
+                return { name: "struct{}", needsPointer: false };
             }
 
-            write("// ");
-            writeLine(line);
+            throw new Error("Unexpected non-empty literal object: " + JSON.stringify(type.value));
+
+        case "or": {
+            return handleOrType(type);
         }
-    }
-}
 
-/**
- * @param {string | undefined} deprecated
- */
-function writeDeprecation(deprecated) {
-    if (deprecated) {
-        writeLine("//");
-        write("// Deprecated: ");
-        writeLine(deprecated);
+        default:
+            throw new Error(`Unsupported type kind: ${type.kind}`);
     }
 }
 
 /**
- * @param {string} s
+ * @param {OrType} orType
+ * @returns {GoType}
  */
-function titleCase(s) {
-    return s.charAt(0).toUpperCase() + s.slice(1);
-}
+function handleOrType(orType) {
+    // Check for nullable types (OR with null)
+    const nullIndex = orType.items.findIndex(item => item.kind === "base" && item.name === "null");
+
+    // If it's nullable and only has one other type
+    if (nullIndex !== -1) {
+        if (orType.items.length !== 2) {
+            throw new Error("Expected exactly two items in OR type for null handling");
+        }
 
-/**
- * @typedef {{ type: Type; name: string; }} UnionMember
- */
-void 0;
+        const otherType = orType.items[1 - nullIndex];
+        const resolvedType = resolveType(otherType);
 
-/**
- * Map from literal value to type name.
- * @type {Map<string | number | boolean, string>}
- */
-const literalTypes = new Map();
+        // Use Nullable[T] instead of pointer for null union with one other type
+        return {
+            name: `Nullable[${resolvedType.name}]`,
+            needsPointer: false,
+        };
+    }
 
-/** @type {Map<string, UnionMember[]>} */
-const unionTypes = new Map();
+    const types = orType.items.filter((_, index) => index !== nullIndex);

```

Comment:
Maybe I'm missing something, but do we need to filter this? We're already dealing with the case where `nullIndex !== -1` above.

Reply:
Ah, you're totally right